debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/SnapshotReaderIT.java

```diff
@@ -446,6 +450,81 @@ public void shouldCreateSnapshotSchemaOnlyRecovery() throws Exception {
             fail("failed to complete the snapshot within 10 seconds");
         }
     }
+
+    @Test
+    public void shouldSnapshotTablesInOrderSpecifiedInTablesWhitelist() throws Exception{
+        config = simpleConfig()
+                .with(MySqlConnectorConfig.TABLE_WHITELIST, "connector_test_ro_(.*).orders,connector_test_ro_(.*).Products,connector_test_ro_(.*).products_on_hand,connector_test_ro_(.*).dbz_342_timetest")
+                .build();
+        context = new MySqlTaskContext(config);
+        context.start();
+        reader = new SnapshotReader("snapshot", context);
+        reader.uponCompletion(completed::countDown);
+        reader.generateInsertEvents();
+
+        // Start the snapshot ...
+        reader.start();
+
+        // Poll for records ...
+        List<SourceRecord> records;
+        LinkedHashSet<String> tablesInOrder = new LinkedHashSet<>();
+        LinkedHashSet<String> tablesInOrderExpected = getTableNamesInSpecifiedOrder("orders", "Products", "products_on_hand", "dbz_342_timetest");
+        KeyValueStore store = KeyValueStore.createForTopicsBeginningWith(DATABASE.getServerName() + ".");
+        SchemaChangeHistory schemaChanges = new SchemaChangeHistory(DATABASE.getServerName());
+        while ((records = reader.poll()) != null) {
+            records.forEach(record -> {
+                VerifyRecord.isValid(record);
+                store.add(record);
+                schemaChanges.add(record);
+                if (Optional.ofNullable(((Struct) record.value()).schema().field("op")).isPresent())
+                    tablesInOrder.add(getTableNameFromSourceRecord.apply(record));
+
+            });
+        }
+        assertThat(tablesInOrder).isEqualTo(tablesInOrderExpected);
```

Comment:
This assertions seems not to verify the order. I modified `tablesInOrderExpected` to contain same items in different order and it passed.